guardrails-scripts/yb-voyager-pg-grant-migration-permissions.sql

```diff
@@ -148,75 +148,126 @@ GRANT pg_read_all_stats to :voyager_user;
 
-    \if :proceed
-        \echo 'Proceeding with the transfer of ownership...'
     \else
-        \echo 'Aborting the script.'
-        \q
-    \endif
-
-    -- Add the original owner of the tables to the group
-    \echo ''
-    \echo '--- Adding Original Owner to Replication Group ---'
-    DO $$
-    DECLARE
-        tableowner TEXT;
-        schema_list TEXT[] := string_to_array(current_setting('myvars.schema_list'), ',');  -- Convert the schema list to an array
-        replication_group TEXT := current_setting('myvars.replication_group');  -- Get the replication group from settings
-    BEGIN
-        -- Generate the GRANT statements and execute them dynamically
-        FOR tableowner IN
-            SELECT DISTINCT t.tableowner
-            FROM pg_catalog.pg_tables t
-            WHERE t.schemaname = ANY (schema_list)  -- Use the schema_list variable
-            AND NOT EXISTS (
-                SELECT 1
-                FROM pg_roles r
-                WHERE r.rolname = t.tableowner
-                    AND pg_has_role(t.tableowner, replication_group, 'USAGE')  -- Use the replication_group variable
-            )
-        LOOP
-            -- Display the GRANT statement
-            RAISE NOTICE 'Granting role: GRANT % TO %;', replication_group, tableowner;
-
-            -- Execute the GRANT statement
-            EXECUTE format('GRANT %I TO %I;', replication_group, tableowner);
-        END LOOP;
-    END $$;
+        -- Option 2: Grant the original owners of the tables to the migration user
+        -- Count rdsadmin-owned tables in selected schemas and expose as psql variable
+        SELECT COUNT(*) AS rdsadmin_owned_table_count
+        FROM pg_catalog.pg_tables t
+        WHERE t.schemaname = ANY(string_to_array(current_setting('myvars.schema_list'), ',')::text[])
+        AND t.tableowner = 'rdsadmin';
+        \gset
+
+        -- If any found, warn per-table and prompt to proceed
+        \if :rdsadmin_owned_table_count
+            \echo ''
+            DO $$
+            DECLARE
+                r RECORD;
+            BEGIN
+                FOR r IN
+                    SELECT t.schemaname, t.tablename
+                    FROM pg_catalog.pg_tables t
+                    WHERE t.schemaname = ANY(string_to_array(current_setting('myvars.schema_list'), ',')::text[])
+                    AND t.tableowner = 'rdsadmin'
+                LOOP
+                    RAISE WARNING 'Table %.% is owned by rdsadmin; cannot grant this role to %.',
+                        r.schemaname, r.tablename, current_setting('myvars.voyager_user');
+                END LOOP;
+            END $$;
+
+            \echo 'Found ' :rdsadmin_owned_table_count ' rdsadmin-owned table(s) in the selected schema(s).'
+            \echo 'These tables are owned by rdsadmin; role membership cannot be granted to the migration user.'
+            \prompt 'Proceed anyway? We will not be granting ownership to the migration user for these tables. (yes/no): ' proceed
```

Comment:
Giving a prompt in this will grant incomplete permissions, right?
Should we abort in this case and ask users to use option 1 only?